pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +1008,58 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// ShouldUpdateZoneEndpointList checks whether zone endpoint list need to be updated or not
+func ShouldUpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) (bool, error) {
+	if zoneName == "" {
+		return false, errors.Errorf("zone name can't be empty")
+	}
+
+	zoneExists, endpoints := findZoneEndpoints(zoneName, zones)
+	if !zoneExists {
+		return false, nil
+	}
+
+	zoneEndpointListEqual, err := listsAreEqual(desiredEndpointList, endpoints)
+	return !zoneEndpointListEqual, err
+}
+
+func findZoneEndpoints(targetZone string, zones []zoneType) (bool, []string) {
+	for _, z := range zones {
+		if z.Name == targetZone {
+			return true, z.Endpoints
+		}
+	}
+	return false, []string{}
+}
+
+func listsAreEqual(a, b []string) (bool, error) {
+	as := make([]string, len(a))
+	bs := make([]string, len(b))
+	copy(as, a)
+	copy(bs, b)
+	sort.Strings(as)
+	sort.Strings(bs)
+
+	if checkDuplicates(as) || checkDuplicates(bs) {
```

Comment:
Could the duplicate entries just be ignored instead of cause an error?

Reply:
@BlaineEXE  proposed erroring out the scenario for duplicate entires present in Zone.CustomEndpoints. There is no functionality impact from RGW multisite point of view even if we ignore duplicate entries. Details mentioned in https://github.com/rook/rook/pull/10629#discussion_r931472999